City-of-Helsinki/drupal-helfi-etusivu
# UHF-9961: Finto AI api client

# [UHF-9961](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9961)


## What was done


* Initial setup for suggestions pilot

## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout UHF-9961-ai-suggestions`
  * `make fresh`
* Run `make drush-cr`

## How to test


* [x] Add few items to recommendations index: `drush sapi-i recommendations --limit 100`.
* [x] Try to query some results:
  * Example query: `curl -X POST -H 'Content-Type: application/json' http://elastic:9200/recommendations/_search -d '{ "size": 10, "query": { "bool": { "filter": [ { "match": { "ai_keywords.label": "vappu" }}]}}}' | jq`
  * Valid keywors are YSO terms: https://finto.fi/yso/fi/
  * Check that translations are in place: `curl -X POST -H 'Content-Type: application/json' http://elastic:9200/recommendations/_search -d '{ "size": 2, "query": { "bool": { "filter": [ { "match": { "ai_keywords.uri": "http://www.yso.fi/onto/yso/p16575" }}, { "match": { "_language": "sv" }}]}}}' | jq`
* [ ] Check that entity to text conversion makes sense. You can use the debugging page at `https://helfi-etusivu.docker.so/fi/admin/content/helfi-integrations/text-preview/node/{id}`. Read the documentation page for more details: https://github.com/City-of-Helsinki/drupal-helfi-etusivu/blob/UHF-9961-ai-suggestions/public/modules/custom/helfi_annif/README.md.

## Continuous documentation


* [ ] This feature has been documented/[the documentation](https://github.com/City-of-Helsinki/drupal-helfi-etusivu/blob/UHF-9961-ai-suggestions/public/modules/custom/helfi_annif/README.md) has been updated

## Other PRs




[UHF-9961]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9961?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
